screens/Providers.js

```diff
@@ -1,26 +1,45 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { StyleSheet, View, FlatList, Button, Text, Modal, TouchableOpacity } from 'react-native';
 import { grey } from 'color-name';
 import { GlobalColors } from '../assets/GlobalStyle';
 import ListTemplate from '../list-template/ListTemplate';
 import ProviderForm from "../form_components/ProviderForm";
 
+import { allSchemas, defaultOpenParams } from '../realm/DatabaseConfig';
+
+const Realm = require('realm');
+
 export default function Providers() {
     
     const [modalOpen, setModalOpen] = useState(false);
+    const [providers, setProvider] = useState(null);
+    const [realm, setRealm] = useState(null);
+
+    useEffect(() => {
+        Realm.open(defaultOpenParams).then(realm => {
+            console.log('opened realm in Treatments');
+            setRealm(realm);
+            setProvider(realm.objects('Provider'));
+        });
 
-    const [providers, setProvider] = useState([
-        { text: "Dr. Oz", field: 'phony', address: 'fakersvile', primary_contact: '555-555-5555', key: '1' },
-        { text: 'Dr. Phil', field: 'phony', address: 'fakersvile', primary_contact: '555-555-5555', key: '2' },
-        { text: 'Dr. Drew', field: 'phony', address: 'fakersvile', primary_contact: '555-555-5555', key: '3' }
-    ]);
+        return () => {
+            if (realm !== null && !realm.isClosed) {
+                console.log('closed realm');
+                realm.close();
+            }
+        };
+    }, []);
 
     //find better way of generating new key
     const addProvider = (providers) => {
-        providers.key = Math.random().toString();
-        setProvider((nextProvider) => {
-            return [providers, ...nextProvider];
-        })
+        realm.write(() => {
+            newTreatment = realm.create('Provider', {
+                firstName: appointments.firstN,
```

Comment:
Yo! Can you break down how we use the providers parameter, how we have access to appointments or treatments vars?